src/panelApi.js

```diff
@@ -6,22 +6,76 @@ import Debug from 'debug'
 
 const debug = Debug('pdk:panelapi')
 
+function parseJwt (token) {
+  return JSON.parse(Buffer.from(token.split('.')[1], 'base64').toString());
+}
+
+function getTokenRefreshInterval (token) {
+  const tokenPayload = parseJwt(token)
+  const tokenLifeTime = tokenPayload.exp - tokenPayload.iat
+  return Math.ceil(0.9 * tokenLifeTime)
+}
+
+function sleep (ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
```

Comment:
nit pick, but there's an impl of this in `client.js`, might be a good candidate for a shared utility import.

A built-in may be an even better option as they are abortable for much better cancellation flows than `clearTimeout`; depending on the minimum node version to target:

Node >=14 <16 there are [promisify specializations for timers](https://nodejs.org/dist/latest-v15.x/docs/api/timers.html#timers_settimeout_callback_delay_args).

On node >=16 there are premade [promise-based timers](https://nodejs.org/dist/latest-v18.x/docs/api/timers.html#timerspromisessettimeoutdelay-value-options).